IA-1940: Improve "user management" permission

bmonjoie opened the pull request:
Split the user management permission into "admin user manager" and "user management"

Related JIRA tickets : IA-1940

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- [x] My migrations file are included
- [x] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- Replace the hardcoded permissions with constants
- Implement permission in `/api/profiles/` as well as in `/api/tasks/create/profilesbulkupdate/`
- Slightly change the UI to take into account the new permission

## How to test
Assign the permission "Users (management)" to a user, optionally, assign an org unit to that user to limit its permission to users within that org unit.
Log in with that user and manage users within its health pyramid.

bmonjoie committed:
Code improvements

Replace hardcoded permissions with constants
Split `ProfilesViewSet.partial_update` method for readability.
Add new `USERS_MANAGED` permission (not used yet)

IA-1940

madewulf commented on hat/menupermissions/models.py:
```diff
@@ -17,46 +17,75 @@
 from django.utils.translation import gettext_lazy as _
 
-MODIFICATIONS = _("Modifications")
+_ASSIGNMENTS = "iaso_assignments"
```
Didn't you remove translations here. _() is used to translate the presentation text of a permission, normally

bmonjoie replied:
I have moved the translation to the list and used the constants for the keys.

madewulf commented on hat/menupermissions/models.py:
```diff
@@ -66,57 +95,57 @@ class CustomPermissionSupport(models.Model):
+        _COMPLETENESS,
+        _ORG_UNITS,
+        _LINKS,
+        _USERS,
```
Here, you could fix another existing ticket by including all the permissions for a new account_user

bmonjoie replied:
Could you link the ticket here?

bmonjoie committed:
Implement `USERS_MANAGED` permission in API

IA-1940

bmonjoie committed:
Add merge migration

IA-1940

bmonjoie committed:
Fix permission was checking for user id

IA-1940

bmonjoie committed:
Add merge migration

IA-1940

bmonjoie committed:
Black 🙃

madewulf commented on iaso/api/profiles.py:
```diff
@@ -15,19 +16,50 @@
+        if pk == request.user.id:
+            raise PermissionDenied(f"User with '{permission.USERS_MANAGED}' cannot edit their own permissions.")
+
+        org_units = OrgUnit.objects.hierarchy(request.user.iaso_profile.org_units.all()).values_list("id", flat=True)
```
I'm questioning this. A user that can modify the permissions of everybody without creating one could be an interesting option too and would maybe match better what we do with org units limitation (basically, you're more powerful if you have no org units assigned than if you have some)

madewulf commented:
Wondering how all this mixes with the user roles that were recently created and the bulk user update too that we just merged. I have the feeling that we would need to rebase this on main an take these questions into consideration

bmonjoie committed:
Improve user management permission

Implement correct permissions for `iaso_users_managed`.
Change UI to hide element not permitted for `iaso_users_managed`.
Replace hardcoded permissions in the front with constants.
Fix screen saying 401 for unauthorized.

IA-1940

bmonjoie committed:
Improve user management permission

Implement correct permissions for `iaso_users_managed`.
Change UI to hide element not permitted for `iaso_users_managed`.
Replace hardcoded permissions in the front with constants.
Fix screen saying 401 for unauthorized.

IA-1940

bmonjoie committed:
MyPy

beygorghor committed:
Merge branch 'main' into feat/IA-1940_user_management_permission

beygorghor committed:
remove unused perms

beygorghor committed:
fix tests

beygorghor committed:
missing migration

beygorghor committed:
mypy

beygorghor committed:
Merge branch 'main' into feat/IA-1940_user_management_permission

SimonDus reviewed (approved):
I have tested it on my local instance and it seems to work as expected. 
Good job !
